src/Application/SqlScripts/SqlJoinScript.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Ravenfire\Magpie\Application\SqlScripts;
+
+use Illuminate\Database\Capsule\Manager as DB;
+use Ravenfire\Magpie\Application\AbstractMagpieCommand;
+use Symfony\Bridge\Monolog\Handler\ConsoleHandler;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+
+class SqlJoinScript extends AbstractMagpieCommand
+{
+    protected static $defaultName = 'sql:join';
+    protected static $defaultDescription = "Sql query counting the number of every group in a column";
+
+    protected function configure(): void
+    {
+        $this->setHelp("Sql query counting the number of every group in a column");
+        $this->addArgument('tableOne', InputArgument::REQUIRED, "First table to use");
+        $this->addArgument('tableTwo', InputArgument::REQUIRED, "Second table to use");
+        $this->addArgument('tableOneJoinColumn', InputArgument::REQUIRED, "Table one column to join");
+        $this->addArgument('tableTwoJoinColumn', InputArgument::REQUIRED, "Table two column to join");
+        $this->addOption('confirm', '-c', InputOption::VALUE_OPTIONAL, 'Confirm?', false);
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->getContext()->getLogger()->pushHandler(new ConsoleHandler($output));
+
+        $tableOne = $input->getArgument('tableOne');
+        $tableTwo = $input->getArgument('tableTwo');
+        $tableOneJoinColumn = $input->getArgument('tableOneJoinColumn');
+        $tableTwoJoinColumn = $input->getArgument('tableTwoJoinColumn');
+
+        $results = $this->index($tableOne, $tableTwo, $tableOneJoinColumn, $tableTwoJoinColumn);
+
+        foreach ($results[0] as $result => $data) {
+            $dbColumns[] = $result;
+        }
+
+        $rows = [];
+        $row = [];
+
+        foreach ($results as $result) {
```

Comment:
It looks like you could extract some of this to a function if you wanted.

You could even insert a class in between `AbstractMagpieCommand` like `SqlCommand` that has some of the shared logic init.

So `AbstractMagpieCommand` > `SqlCommand` > Each of these classes.

The `SqlCommand` could be abstract or not. I don't think it matters since you probably won't need any "must be implemented" methods, but only use `SqlCommand` to hold some functions that all 3 of your scripts could use.

Just a suggestion.

Reply:
Or, you could explore php `Trait`s, which are a different (related) way to share code between different files.